src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/util/IndexManagementException.kt

```diff
@@ -0,0 +1,45 @@
+package com.amazon.opendistroforelasticsearch.indexmanagement.util
+
+import org.elasticsearch.ElasticsearchException
+import org.elasticsearch.common.Strings
+import org.elasticsearch.common.ValidationException
+import org.elasticsearch.index.IndexNotFoundException
+import org.elasticsearch.rest.RestStatus
+import java.lang.IllegalArgumentException
+
+class IndexManagementException(message: String, val status: RestStatus, ex: Exception) : ElasticsearchException(message, ex) {
```

Comment:
Are there any issues with serialization/deserialization of this exception class? I thought Bukhtawar or someone brought that up before.

Reply:
for the IndexManagementException, currently is only used like this `actionListener.onFailure(...)`
so I think it doesn't involve inter-node communication